packages/ra-core/src/controller/list/useListController.ts

```diff
@@ -219,7 +219,7 @@ export interface ListControllerResult<RecordType extends RaRecord = any> {
     onUnselectItems: () => void;
     page: number;
     perPage: number;
-    refetch: UseGetListHookValue<RecordType>['refetch'];
+    refetch: (() => void) | UseGetListHookValue<RecordType>['refetch'];
```

Comment:
I think we should remove the `(() => void)`

Reply:
Done to  continue the compatibility between `ListContext` and `ChoicesContext` for `DatagridInput`